File: src/DynamicLinkLibraries/Diagram/Diagram/Interfaces/IFeedbackAliasCollection.cs

using System.Collections.Generic;

namespace Diagram.UI.Interfaces
{
    /// <summary>
    /// Feedback alias connection
    /// </summary>
    public interface IFeedbackAliasCollection
    {
        /// <summary>
        /// Dictionary
        /// </summary>
        Dictionary<string, string> Dictionary { get; }
        
        /// <summary>
        /// Aliases
        /// </summary>
        IEnumerable<IFeedbackAlias> Aliases { get; }

        /// <summary>
        /// Adds alias
        /// </summary>
        /// <param name="alias">The alias</param>
        void Add(IFeedbackAlias alias);

        /// <summary>
        /// Fills itself
        /// </summary>
        void Fill();

        /// <summary>
        /// Holder
        /// </summary>
        IFeedbackAliasCollectionHolder Holder { get; }

        /// <summary>
        /// Sets itself
        /// </summary>
        void Set();


    }
}

File: src/TypeScript/TypeScriptLibrary/Library/Measurements/Interfaces/IFeedbackAliasCollection.ts
import { IFeedbackAlias } from "../../Interfaces/IFeedbackAlias";

export interface IFeedbackAliasCollection {
    addFeedbackAlias(feedbackAlias: IFeedbackAlias): void;
    updateFeedbackAliasCollection(): void;
    getExternalAliasDictionary(): Map<string, string>;
    getExternalAlias(name: string): string;
    addExternalAlias(name: string, value: string): void;
}